### ISSUE TITLE
Android TV fails to play certain formats

### ISSUE THREAD

**Issue by Jarr-ett** [jellyfin/jellyfin-androidtv#53]
Whenever I try to play any media on an Nvidia Shield it says "Failed to load content"

Server errors when trying to play:
```
Feb 19 14:21:04 jelly jellyfin[17612]: [14:21:04] [ERR] AcceptWebSocketAsync error
    Feb 19 14:21:04 jelly jellyfin[17612]: SocketHttpListener.WebSocketException: net_WebSockets_AcceptUnsupportedWebSocketVersion
    Feb 19 14:21:04 jelly jellyfin[17612]:    at SocketHttpListener.Net.WebSockets.HttpWebSocket.ValidateWebSocketHeaders(HttpListenerContext context) in /jellyfin/SocketHttpListener/Net/WebSockets/HttpWebSocket.cs:line 155
    Feb 19 14:21:04 jelly jellyfin[17612]:    at SocketHttpListener.Net.WebSockets.HttpWebSocket.AcceptWebSocketAsyncCore(HttpListenerContext context, String subProtocol, Int32 receiveBufferSize, TimeSpan keepAliveInterval, Nullable`1 internalBuffer) in /jellyfin/SocketHttpListener/Net/WebSockets/HttpWebSocket.Managed.cs:line 23
    Feb 19 14:21:04 jelly jellyfin[17612]:    at Jellyfin.Server.SocketSharp.WebSocketSharpListener.ProcessWebSocketRequest(HttpListenerContext ctx) in /jellyfin/Jellyfin.Server/SocketSharp/WebSocketSharpListener.cs:line 151
```
```
    Feb 19 14:21:24 jelly jellyfin[17612]: [14:21:24] [ERR] Error processing request
    Feb 19 14:21:24 jelly jellyfin[17612]: MediaBrowser.Common.Extensions.ResourceNotFoundException: Nickelodeon does not have an image of type Primary
    Feb 19 14:21:24 jelly jellyfin[17612]:    at MediaBrowser.Api.Images.ImageService.GetImage(ImageRequest request, Guid itemId, BaseItem item, Boolean isHeadRequest) in /jellyfin/MediaBrowser.Api/Images/ImageService.cs:line 572
    Feb 19 14:21:24 jelly jellyfin[17612]:    at Emby.Server.Implementations.Services.ServiceExecGeneral.Execute(Type serviceType, IRequest request, Object instance, Object requestDto, String requestName) in /jellyfin/Emby.Server.Implementations/Services/ServiceExec.cs:line 88
    Feb 19 14:21:24 jelly jellyfin[17612]:    at Emby.Server.Implementations.Services.ServiceController.Execute(HttpListenerHost appHost, Object requestDto, IRequest req) in /jellyfin/Emby.Server.Implementations/Services/ServiceController.cs:line 177
    Feb 19 14:21:24 jelly jellyfin[17612]:    at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost appHost, IRequest httpReq, IResponse httpRes, ILogger logger, String operationName, CancellationToken cancellationToken) in /jellyfin/Emby.Server.Implementations/Services/ServiceHandler.cs:line 95
    Feb 19 14:21:24 jelly jellyfin[17612]:    at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken) in /jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 608
```
```
    Feb 19 14:21:27 jelly jellyfin[17612]: [14:21:27] [ERR] Error this.ProcessRequest(context)(Exception while writing error to the response)
    Feb 19 14:21:27 jelly jellyfin[17612]: System.InvalidOperationException: Operation is not valid due to the current state of the object.
    Feb 19 14:21:27 jelly jellyfin[17612]:    at SocketHttpListener.Net.HttpListenerResponse.set_ContentLength64(Int64 value) in /jellyfin/SocketHttpListener/Net/HttpListenerResponse.cs:line 89
    Feb 19 14:21:27 jelly jellyfin[17612]:    at Emby.Server.Implementations.HttpServer.HttpListenerHost.Write(IResponse response, String text) in /jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 688
    Feb 19 14:21:27 jelly jellyfin[17612]:    at Emby.Server.Implementations.HttpServer.HttpListenerHost.ErrorHandler(Exception ex, IRequest httpReq, Boolean logExceptionStackTrace, Boolean logExceptionMessage) in /jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 229`
```

Server info
 - OS: Debian 9
 - Jellyfin Version: 10.2.0 from PPA

Client info
- Nvidia Shield 
- jellyfin-androidtv_google-release_beta3.apk 

Addition info
Streaming works with the regular Android app on the Shield.
According to other users, it only seems to be a problem on Debian servers (works with Windows server)

**Comment by flex36ty** [comment]
If it was the app why would it work 100% when playing from a windows server? This is only happening to Linux servers. I've had zero playback issues with android tv and windows server

**Comment by Jarr-ett** [comment]
> If you can provide any additional information about the types of media that you are trying to play, that could help narrow this down.

The most basic media type I've tried (which fails to play) was  h264, 720p, AAC, no subtitles.
Also fails with h265, AAC, with/without subtitles

**Comment by Jarr-ett** [comment]
I was wrong. Found an MPEG4, MP3 video and it plays fine.

**Comment by bfayers** [comment]
> I was wrong. Found an MPEG4, MP3 video and it plays fine.

This might be related to AAC audio, perhaps.

Could you re-title the issue, especially now that it's clear that the androidtv app *will* play some things.

**Comment by flex36ty** [comment]
It's a bug in the Android TV app. I downloaded an older version of the emby app and tested playing from my Linux server which played everything successfully

**Comment by Jarr-ett** [comment]
> This might be related to AAC audio, perhaps.

Also fails with h.264, AC3
That's the limits of media library, so I can't narrow it down anymore...

```
Fails:
h.264/AC3
h.264/AAC
h.265/AAC
```
```
Works:
MPEG4/MP3
```

**Comment by thornbill** [comment]
We just released a version of the app (beta 4) which includes an update to the base version of the app. These changes resolved a lot of the playback issues that I was seeing in my testing. Please test in the latest release and let us know if your playback issues are also resolved. Thanks!

**Comment by Jarr-ett** [comment]
Working great now. Playing everything now! Thanks

**Comment by thornbill** [comment]
Fixed by #58.
